src/client/utils.ts

```diff
@@ -27,6 +27,8 @@ export function doesLocalStorageMatch(newValue: string | null, user: UserFromTok
         return false
     }
 
+    removeUndefinedValues(user)
```

Comment:
It's typically easier to reason about systems if the objects are immutable - especially something like UserFromToken which is used in a lot of different places.

A simple way to update this is to do something like:
```
return isEqual(parsed, jsonSerialize(user))
```

where `jsonSerialize` is just
```
// We need to make sure that the comparison is done with objects that have gone through the same transformation, so we mimic the localStorage transformation to json and back
function jsonSerialize(userFromToken: UserFromToken) {
  return JSON.parse(JSON.stringify(userFromToken))
}
```

that way the object isn't actually changed